### DISCUSSION TITLE
How to refer response header in client generate api

### DISCUSSION THREAD

**Discussion by Radha-rog** [OAI/OpenAPI-Specification#2815] [Q&A]
I have defined the swagger in order to get the authorization token from the response header which can be  used for next api call. However when I generate the client code out of swagger, it return only response body and not the header. I Have tried all the search but nothing was helpful. 

**Part of the swagger definition,**

```yaml
/token':
    post:
      tags:
        - Token
      parameters:
        - $ref: '#/components/parameters/Authorization'
      description: Used to generate token to call other api 
      operationId: generateToken
responses:
        '200':
          description: OK
          content:
            Application/Json:
              schema:
                type: array
                items:
                  $ref: '#/components/schemas/XYZ'
          headers:
            authorization:
              schema:
                type: string

components:
  parameters:
    Authorization:
      in: header
      name: Authorization
      description: Bearer token to access the one view API's.
      schema:
        type: string
      required: true
```
 
**Client Api generate:** ( the method body doesnt have any code for response header"
```java
public CompletableFuture<List<XYZ>> generateToken (String authorization) throws ApiException {
....
...}
```
If you could provide any insight that would be more helpfull

**Comment by webron** [comment]
In general, you cannot define `Authorization` as a header parameter and you have to use [Security Scheme](https://spec.openapis.org/oas/v3.1.0.html#security-scheme-object) to define it.

That said, if you have an issue with the generated code, you'd need to file an issue in repository of the project you use. This repository is meant only for issues related to spec itself.

**Comment by Radha-rog** [comment]
@webron 

Below are the changes made as per your suggestions. Still its not resolved. 
```yaml
responses:
        '200':
          description: OK
          headers:
            authorization:
              schema:
                $ref: '#/components/securitySchemes/BearerAuth'
              description: Oneview Api token

components:
  securitySchemes:
    BearerAuth:
      type: http
      scheme: bearer
```
Moreover, there is no response for this api, so the generated client api is throwing exception at catch block since return value is null. how can that be fixed?
```java
try {
                  return CompletableFuture.completedFuture(
                          null
                  );
              } catch (IOException e) {
                  return CompletableFuture.failedFuture(new ApiException(e));
              }
```
If this is not the right repository, can you tell me the repository where I can raise the issue?

My sample client generator config,

```xml
 <execution>
    <id>XXX</id>
    <goals>
        <goal>generate</goal>
    </goals>
    <configuration>
        <inputSpec>
            ${project.basedir}/src/main/resources/swagger-spec/client/XXX-api.yaml
        </inputSpec>

        <generatorName>java</generatorName>
        <library>native</library>
        <output>${project.basedir}</output>

        <generateApis>true</generateApis>
        <generateApiTests>false</generateApiTests>
        <generateApiDocumentation>false</generateApiDocumentation>
        <generateModels>true</generateModels>
        <generateModelTests>false</generateModelTests>
        <generateModelDocumentation>false</generateModelDocumentation>
        <generateSupportingFiles>true</generateSupportingFiles>
        <apiPackage>XYZ</apiPackage>
        <invokerPackage>XYZ</invokerPackage>
        <modelPackage>XYZ</modelPackage>
        <addCompileSourceRoot>true</addCompileSourceRoot>

        <configOptions>
            <java8>true</java8>
            <dateLibrary>java8</dateLibrary>
            <delegatePattern>true</delegatePattern>
            <useBeanValidation>true</useBeanValidation>
            <performBeanValidation>true</performBeanValidation>
            <configPackage>XYZ</configPackage>
            <interfaceOnly>true</interfaceOnly>
            <sortParamsByRequiredFlag>false</sortParamsByRequiredFlag>
            <debugOperations>true</debugOperations>
            <debugModels>true</debugModels>
            <debugOpenAPI>true</debugOpenAPI>
            <asyncNative>true</asyncNative>
            <debugSupportingFiles>true</debugSupportingFiles>
        </configOptions>
    </configuration>
</execution>
```
Please guide to resolve the issue as its blocking my work

**Reply by webron** [reply]
That's still not the right way to describe it, you can refer to https://swagger.io/docs/specification/authentication/ for further explanation. I can't tell you which repository is the right one to post the issue with the generated code because I don't know which generator you're using.

**Comment by Radha-rog** [comment]
@webron, My swagger worked without making any change even the old code definition for header worked. Found the fix from the issues reported on openAPItool and  the fix was provided on 5.2.0. After using that version it worked. Client code regenerated to return headers.

Thank you very much for your time and support

**Comment by handrews** [comment]
Issue resolved, closing.
